### DISCUSSION TITLE
Disabled SIP and installed scripting addition, but still getting error

### DISCUSSION THREAD

**Discussion by farzadmf** [asmvik/yabai#1777] [General]
So, I finally decied to give SIP and scripting addition a shot (main driver: to be able to work with spaces).

I disabled SIP with this command [from the Wiki](https://github.com/koekeishiya/yabai/wiki/Disabling-System-Integrity-Protection)
```
csrutil disable --with kext --with dtrace --with nvram --with basesystem
```

First question: when running this, it warns me about:
```
requesting an unsupported configuration. This is likely to break in the future and leave your machine in an unknown state.
```

which is scary, and I thin it implies that there's the possibility of me updating the OS and my Mac dying and not be usable, is that the case?

---

Other than that, with SIP being disabled, I followed [instructions from the Wiki](https://github.com/koekeishiya/yabai/wiki/Installing-yabai-(latest-release)#configure-scripting-addition) to install scripting additions:
- Added the `sudoer` file
- Updated `yabairc` with the lines

But when I run a space-focus command; e.g. `yabai -m space --focus 2`, I get:
```
cannot focus space due to an error with the scripting-addition.
```

which is the exact error I get before disabling SIP and other steps.

I tried running `sudo yabai --install-sa` and the command runs, but I don't think anything happens because I still get the same error.

It goes without saying that I have restarted yabai using `yabai --restart-service` multiple times, but nothing changed

**Comment by gldtn** [comment]
Sounds like your /etc/sudoer file may not been configured correctly or it has not been loaded with yabairc

Run 
```
sudo cat /private/etc/sudoers.d/yabai
```

And make sure it matches the output of:
```
echo "$(whoami) ALL=(root) NOPASSWD: sha256:$(shasum -a 256 $(which yabai) | cut -d " " -f 1) $(which yabai) --load-sa"
```

If not, Copy the output of the echo command above to the `/etc/sudoers` and save the file.

Also I know you said you did this, but triple check these are the exact lines at the top of yabairc:
```
yabai -m signal --add event=dock_did_restart action="sudo yabai --load-sa"
sudo yabai --load-sa
```

Finally restart yabai.. not sure if it actually requires a computer restart!

**Comment by gldtn** [comment]
> which is scary, and I thin it implies that there's the possibility of me updating the OS and my Mac dying and not be usable, is that the case?

I've been running Yabai for some time now, and also I just updated to Sonoma without any problems at all. If you do run into problems with updates, you can always enable SIP again, update MacOS and disable it again!

**Comment by gldtn** [comment]
> I disabled SIP with this command [from the Wiki](https://github.com/koekeishiya/yabai/wiki/Disabling-System-Integrity-Protection)
> 
> ```
> csrutil disable --with kext --with dtrace --with nvram --with basesystem
> ```

On last thing, which version of MacOS are you on? I'm assuming your running Intel based system with the command above!

If your on APPLE SILICON, you need to use the correct SIP command depending on the version of you MacOS.

**Comment by farzadmf** [comment]
Thanks for all the answers @gldtn . It's been a while that I posted this, and I also posted this as an "issue" in the repo, and my issue was resolved, so, at least for now, things to be working fine

---

EDIT: since my issue, as I mentioned, is resolved, I'm gonna close the discussion
